src/haz3ltest/Test_Elaboration.re

```diff
@@ -0,0 +1,290 @@
+open Alcotest;
```

Comment:
Looking at other ocaml projects it looks standard to put your tests in `test/. I'm not sure if it's worth doing that sooner rather than later if we want to avoid causing a ton of merge conflicts. @cyrus- do you have thoughts?